src/main/java/dp/handler/Handler.java

```diff
@@ -122,18 +122,26 @@ public void listen(final ExportedFile message) {
         MessageType messageType = GetMessageType(message);
 
         try {
-
+            Integer rowCount=0;
+            if (message != null && message.getRowCount() != null){
```

Comment:
Sorry I hadn't spotted this before, I think we need to set rowCount to message.getRowCount() when it is initialised otherwise it will have value 0 in the if statements further down. So lines 125 and 126 probably need to look like this:
```suggestion
            Integer rowCount = message.getRowCount();
            if (rowCount != null){
```
NB. There seems to be no point in checking for message != null here because the code would have failed with a null pointer anyway if that were the case.

Reply:
This has now been changed